From 8c9ee7f5753d94a2f2760ed236ae4b527c7e1688 Mon Sep 17 00:00:00 2001
From: Julien <julien@caffeine.lu>
Date: Fri, 11 Sep 2026 10:30:09 +0000
Subject: [PATCH] Use `assertTemplateContains` in `pulumi new` tests (#24639)

The column spacing depends on the names of the templates, which can
change underneath us. We must always use the `assertTemplateContains` to
guard against this.
---
 pkg/cmd/pulumi/project/newcmd/new_test.go | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/pkg/cmd/pulumi/project/newcmd/new_test.go b/pkg/cmd/pulumi/project/newcmd/new_test.go
index 054fa675ba7d..a76092fe3053 100644
--- a/pkg/cmd/pulumi/project/newcmd/new_test.go
+++ b/pkg/cmd/pulumi/project/newcmd/new_test.go
@@ -1078,7 +1078,7 @@ func TestPulumiNewWithOrgTemplates(t *testing.T) {
 Available Templates:
 `)
 	// Check that our org based templates are there
-	assert.Contains(t, stdout.String(), `
+	assertTemplateContains(t, stdout.String(), `
   template-1                         Describe 1
   template-2                         Describe 2
 `)
@@ -1140,9 +1140,12 @@ func TestPulumiNewWithRegistryTemplates(t *testing.T) {
 	assert.Contains(t, stdout.String(), `
 Available Templates:
 `)
+
 	// Check that our registry based templates are there with the appropriate disambiguation prefix.
-	assert.Contains(t, stdout.String(), "template-1 [Some org]              Describe 1")
-	assert.Contains(t, stdout.String(), "template-2 [some-org/repo]         Describe 2")
+	assertTemplateContains(t, stdout.String(), `
+template-1 [Some org]              Describe 1
+template-2 [some-org/repo]         Describe 2
+`)
 
 	// Check that normal templates are there
 	assertTemplateContains(t, stdout.String(), `
@@ -1223,6 +1226,8 @@ func TestPulumiNewWithoutTemplateSupport(t *testing.T) {
 	// Check that normal templates are there
 	assert.Contains(t, stdout.String(), `
 Available Templates:
+`)
+	assertTemplateContains(t, stdout.String(), `
   aiven-go                           A minimal Aiven Go Pulumi program
 `)
 	assert.Equal(t, "", stderr.String())
